__tests__/integration/utils/wallet-helper.js

```diff
@@ -146,18 +155,42 @@ export class WalletHelper {
       walletsArr.unshift(new WalletHelper(genesis.walletId, { words: genesis.words }));
     }
 
-    // Requests the start of all the wallets in quick succession
-    const startPromisesArray = [];
-    for (const wallet of walletsArr) {
-      const promise = TestUtils.startWallet(wallet.walletData);
-      walletsPendingReady[wallet.walletId] = wallet;
-      startPromisesArray.push(promise);
+    // Start of the requests
+    startBenchmark.requestsStart = Date.now().valueOf();
+
+    if (options.serial || walletsArr.length > 2) {
+      // If we need to initialize too many wallets at once, it's better to do it serially
+      for (const wallet of walletsArr) {
+        const walletBenchmark = {};
+        walletBenchmark.requestStart = Date.now().valueOf();
+        await TestUtils.startWallet(wallet.walletData, {
+          waitWalletReady: true
+        });
+        walletBenchmark.requestEnd = Date.now().valueOf();
+        walletsPendingReady[wallet.walletId] = wallet;
```

Comment:
Why are you adding to the pending ready object if you are using `waitWalletReady: true`?

Reply:
This is for redundancy.

On my machine, once the call with `waitWalletReady: true` is made, the wallet is in fact ready to be used. But on GitHub CI this was not happening.

I tried other delays and solutions, but they could add too much complexity to this code that is already hard to read.

So I decided on a simple redundancy and used existing code to double-check for the wallet status to be `Ready`. It looks odd, but it's simple to understand.

What do you think?

Reply:
I think there might be something wrong with `waitWalletReady` parameter if it's not working on CI. Maybe we shouldn't have it if we can't trust that will work as expected.

Reply:
In fact there was an obsolete verification on the wallet being ready. With the performance improvements it became buggy and was removed on [015a196](https://github.com/HathorNetwork/hathor-wallet-headless/pull/164/commits/015a196b41e3121b24fc8073e0ee13c2c4a801e0).

Another step to simplify this process :+1: